src/Chart.php

```diff
@@ -112,68 +156,80 @@ public function setOptions(array $options)
         return $this;
     }
 
+    /**
+     * @param array<string, array<mixed, mixed>> $options column_name => array of options
+     *
+     * @return $this
+     */
+    public function setColumnOptions(array $options)
+    {
+        // IMPORTANT: use replace not merge here to preserve numeric keys !!!
+        $this->columnOptions = array_replace_recursive($this->columnOptions, $options);
```

Comment:
`array_replace_recursive` function is highly problematic, see https://github.com/atk4/ui/issues/1811 and related data bug

why we need it and do you understand all the side effects? I prefer to never use this function across all atk4 repos

Reply:
Not sure about all side effects, but replace was needed here to be able to merge array elements with numeric keys. array_merge_recursive appends (not merge/replace) elements with numeric keys.